chain/network/src/peer_manager/peer_store/mod.rs

```diff
@@ -195,13 +205,22 @@ impl Inner {
         self.addr_peers
             .insert(peer_addr, VerifiedPeer { peer_id: peer_info.id.clone(), trust_level });
 
-        let now = clock.now_utc();
-
-        // Update peer_id addr
-        self.peer_states
-            .entry(peer_info.id.clone())
-            .and_modify(|peer_state| peer_state.peer_info.addr = Some(peer_addr))
-            .or_insert_with(|| KnownPeerState::new(peer_info.clone(), now));
+        // Update or insert peer_id addr
+        if let Some(peer_state) = self.peer_states.peek_mut(&peer_info.id) {
```

Comment:
please check whether we actually need to keep both peer_stats and addr_peers at this point. Perhaps this can be simplified.